tests/base/src/test/java/org/wildfly/security/http/HttpAuthenticatorTest.java

```diff
@@ -111,11 +111,15 @@ public void testDigestMd5() throws Exception {
     public void testBasic() throws Exception {
         testOneOfThree();
 
+        Assert.assertFalse(authenticator.authenticate());
+        List<String> responses = exchangeSpi.getResponseAuthenticateHeaders();
+        assertEquals("Basic authentication sends challenge when AUTHORIZATION header is present", 6, responses.size());
+        Assert.assertEquals(UNAUTHORIZED, exchangeSpi.getStatusCode());
+
         exchangeSpi.setRequestAuthorizationHeaders(Collections.singletonList(
                 "Basic TXVmYXNhOkNpcmNsZSBvZiBMaWZl"
         ));
         Assert.assertTrue("Basic successful", authenticator.authenticate());
-        Assert.assertEquals(0, exchangeSpi.getStatusCode());
```

Comment:
@RanabirChakraborty Thanks for the PR! I would rather not remote this line since it did test the response before. New test can be added instead. But I noticed that something similar is tested here already: https://github.com/wildfly-security/wildfly-elytron/blob/1.x/tests/base/src/test/java/org/wildfly/security/http/HttpAuthenticatorTest.java#L105 . On the other hand I don't see FORBIDDEN response tested for BASIC, so maybe that one could be added instead.

Reply:
You can for example add a test which makes sure that the challenge for HTTP Basic authentication specifies the correct realm, or some other test which is not present currently and relates to basic authentication from RFC2617

Reply:
@RanabirChakraborty  This issue is older and some tests for  BASIC have been added to this class since, so let us know if this issue is out of date now.

Reply:
@Skyllarr sorry for the late, somehow I have missed the notification. I'll check it.